### PR TITLE
[feature-wip](parquet-reader) update column read model and add page index

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/vparquet_reader.cpp [hunk 1]
```diff
@@ -90,63 +95,240 @@ Status ParquetReader::_init_read_columns(const std::vector<SlotDescriptor*>& tup
...
+        *eof = true;
+    }
+    bool _batch_eof = false;
+    auto row_group_reader = _row_group_readers[_current_row_group_id];
+    RETURN_IF_ERROR(row_group_reader->next_batch(block, _batch_size, &_batch_eof));
+    if (_batch_eof) {
+        _current_row_group_id++;
+        if (_current_row_group_id > _total_groups) {
+            *eof = true;
+        }
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_reader.cpp:115]: missing `else`?
In `else` block, you should open the next row group reader and call `next_batch()`

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 2]
```diff
@@ -24,235 +24,57 @@
...
-        auto& column_with_type_and_name = block->get_by_name(read_col.slot_desc->col_name());
-        RETURN_IF_ERROR(_column_readers[read_col.slot_desc->id()]->read_column_data(
-                row_group, &column_with_type_and_name.column));
-        VLOG_DEBUG << column_with_type_and_name.name;
+        auto slot_desc = read_col.slot_desc;
+        auto& column_with_type_and_name = block->get_by_name(slot_desc->col_name());
+        auto column_ptr = column_with_type_and_name.column;
+        auto column_type = column_with_type_and_name.type;
+        RETURN_IF_ERROR(_column_readers[slot_desc->id()]->read_column_data(
+                column_ptr, column_type, batch_size, &_read_rows, _batch_eof));
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:74]: `_read_rows` should be a cumulative value. But here it just be set to the number values of current batch.

File: be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp [hunk 3]
```diff
@@ -43,11 +44,12 @@ Status ColumnChunkReader::init() {
...
     RETURN_IF_ERROR(get_block_compression_codec(_metadata.codec, _block_compress_codec));
 
+    VLOG_DEBUG << "initColumnChunkReader finish";
     return Status::OK();
 }
 
 Status ColumnChunkReader::next_page() {
-    RETURN_IF_ERROR(_page_reader->next_page());
+    RETURN_IF_ERROR(_page_reader->next_page_header());
     _num_values = _page_reader->get_page_header()->data_page_header.num_values;
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp:53]: Better rename `_num_values` to `_left_num_values`.

File: be/src/vec/exec/format/parquet/vparquet_column_reader.cpp [hunk 4]
```diff
@@ -19,50 +19,82 @@
...
+        // seek to next page header
+        _chunk_reader->next_page();
+        if (_row_ranges->size() != 0) {
+            _skipped_pages();
+        }
+        // load data to decoder
+        _chunk_reader->load_page_data();
+    }
+    size_t read_values =
+            _chunk_reader->num_values() < batch_size ? _chunk_reader->num_values() : batch_size;
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_column_reader.cpp:82]: I think we don't need to check the number of values of a `_chunk_reader`.
Just call `_chunk_reader->decode_values` and return the number of values read.

File: be/src/vec/exec/format/parquet/vparquet_reader.cpp (hunk 1 above)
Comment (wsjz) [reply be/src/vec/exec/format/parquet/vparquet_reader.cpp:115]: The `_current_row_group_id++;` statement can do this，so it's implicit to mark `*eof = true`

File: be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp (hunk 3 above)
Comment (wsjz) [reply be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp:53]: remaining

File: be/src/vec/exec/format/parquet/vparquet_column_reader.cpp (hunk 4 above)
Comment (wsjz) [reply be/src/vec/exec/format/parquet/vparquet_column_reader.cpp:82]: default size of a block is `batch_size`, I think we should limit it

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**